Вторая лабораторная работа выполнена

PopovMatvey opened the pull request:
Вторая лабораторная работа выполнена:
1) Выполнены все задания от 1 до 6;
2) В заданиях 1 - 6 выполнены линтеры;
3) В заданиях 1 -6 выполнены все тесты.

PopovMatvey committed:
Вторая лабораторная работа выполнена

Вторая лабораторная работа выполнена:
1) Выполнены все задания от 1 до 6;
2) В заданиях 1 - 6 выполнены линтеры;
3) В заданиях 1 -6 выполнены все тесты.

djilua1 commented on src/ex2_js-basics-part2/task-01.js:
```diff
@@ -0,0 +1,23 @@
+// ## Задание 1
```
Давай без описаний, у нас все описано в ридми.

djilua1 commented on src/ex2_js-basics-part2/task-01.js:
```diff
@@ -0,0 +1,23 @@
+
+const PASSED_VALUE = '';
+
+getSimpleTypeArgument(PASSED_VALUE);// eslint-disable-line no-use-before-define
```
Тестовые данные в реквесте не нужны.

djilua1 commented on src/ex2_js-basics-part2/task-01.js:
```diff
@@ -0,0 +1,23 @@
+
+getSimpleTypeArgument(PASSED_VALUE);// eslint-disable-line no-use-before-define
+
+module.exports = getSimpleTypeArgument;// eslint-disable-line no-use-before-define
```
НА такое правильно линтер ругается, перенеси в конец файла.

djilua1 commented on src/ex2_js-basics-part2/task-01.js:
```diff
@@ -0,0 +1,23 @@
+function getSimpleTypeArgument(argumentValue) {
+  if ((typeof argumentValue === 'number') && !(Number.isNaN(argumentValue))) {
+    return 'number';
+  } if (typeof argumentValue === 'string') {
```
Тут не else is, так что пустой строкой отбил и пишешь.

djilua1 commented on src/ex2_js-basics-part2/task-01.js:
```diff
@@ -0,0 +1,23 @@
+    return 'number';
+  } if (typeof argumentValue === 'string') {
+    return 'string';
+  } if ((typeof argumentValue !== 'number') && (Number.isNaN(argumentValue)) && (typeof argumentValue !== 'string')) {
```
Это условие уже излишне, так как он и так undefined вернет.

djilua1 commented on src/ex2_js-basics-part2/task-01.js:
```diff
@@ -0,0 +1,23 @@
+  }
+
+  return undefined;
+}
```
В последующих тоже сотри описание и тестовые данные.

djilua1 commented on src/ex2_js-basics-part2/task-03.js:
```diff
@@ -0,0 +1,49 @@
+  const returedArray = [];
+
+  for (let i = 0; i < array.length; i++) {
+    if ((typeof array[i] !== 'number') || (Number.isNaN(array[i]))) {
```
Можешь заранее фильтрануть, а так эту проверку можно опустить. Так задание верно сделано.

djilua1 commented on src/ex2_js-basics-part2/task-04.js:
```diff
@@ -0,0 +1,27 @@
+    } else {
+      return false;
+    }
+  }
```
Тут все continue лишние, просто проверь, что все одинаковые. Можно проще сделать.

djilua1 commented on src/ex2_js-basics-part2/task-05.js:
```diff
@@ -0,0 +1,22 @@
+module.exports = getMaxValueArray;// eslint-disable-line no-use-before-define
+
+function getMaxValueArray(array) {
+  let maxValue = -999999999;
```
Лучше взять первый элемент, а в цикле начать со второго, так проще и правильнее.

djilua1 commented on src/ex2_js-basics-part2/task-06.js:
```diff
@@ -0,0 +1,58 @@
+
+  if ((value > MAX_VALUE) || (value < 0)) {
+    return 'Данные неверны';
+  } if ((value === 1) || (value === 0)) {
```
Этот if тоже с новой строки с отступом.

djilua1 commented on src/ex2_js-basics-part2/task-06.js:
```diff
@@ -0,0 +1,58 @@
+    return 'Не причисляется ни к простым, ни к составным числам';
+  }
+
+  for (let i = 0; i < value + 1; i++) {
```
Можно просто с 2 начать и дальше на 0 проверять.

djilua1 commented on src/ex2_js-basics-part2/task-06.js:
```diff
@@ -0,0 +1,58 @@
+
+    return returnedSting;
+  } if (amountDivide > 2) {
+    returnedSting = `Число ${value} - составное число`;
```
В обоих условиях можно просто сразу строку вернуть. `return  'Число ${value} - простое число'`

PopovMatvey committed:
Исправления по второй лабораторной работе

Было исправлено 10 замечаний из 10:
1) Во всех заданиях у меня были комментарии в виде текста задания - сейчас все комментарии были удалены;
2) Были тестовые данные - оставил только функции и экспорт функций;
3) Перенёс экспорт функции в конец файла;
4) В первом задании отбил if пустой строкой;
5) В первом задании удалил лишний if;
6) В четвёртом задании убрал лишние continue и сделал в целом реализацию проще;
7) В пятом задании вместо того, чтобы сравнивать с объявленным числом, начал цикл со второго элемента и сравнивал только элементы между собой;
8) В шестом задании if также отбил пустой строкой;
9) В шестом задании согласно совету начал проверять с 2-ки;
10) В шестом задании убрал в теле условия присвоение к строке и стал возвращать только строку подобным образом : "return `Число ${value} - составное число`;"

Тесты и линтеры, также, все выполнены.
